### PR TITLE
Fix #651(Mythril raises a KeyError for a contract)

### LINKED ISSUE #651: Mythril raises a KeyError for a contract
## Description
Mythril raises a KeyError when run with contract `0x8601e58163C2713B06b40a2000A30FF4A988e926`

## How to Reproduce
```console
$ myth -xila 0x8601e58163C2713B06b40a2000A30FF4A988e926
Traceback (most recent call last):
  File "./myth", line 9, in <module>
    mythril.interfaces.cli.main()
  File "/home/nikhil/MYTHRIL/mythril/mythril/interfaces/cli.py", line 397, in main
    max_transaction_count=args.max_transaction_count,
  File "/home/nikhil/MYTHRIL/mythril/mythril/mythril.py", line 470, in fire_lasers
    issues = fire_lasers(sym, modules)
  File "/home/nikhil/MYTHRIL/mythril/mythril/analysis/security.py", line 20, in fire_lasers
    issues += module.execute(statespace)
  File "/home/nikhil/MYTHRIL/mythril/mythril/analysis/modules/exceptions.py", line 44, in execute
    solver.get_transaction_sequence(state, node.constraints)
  File "/home/nikhil/MYTHRIL/mythril/mythril/analysis/solver.py", line 109, in get_transaction_sequence
    txs[tx_id]["caller"] = caller
KeyError: 'storage_KECCAC[1]'

...
$
```

## Expected behavior
Contract should be executed without any errors

## Environment

- Mythril version: develop
- Solidity compiler and version: `0.4.25`
- Python version: `3.6`
- OS and Version: Ubuntu 17.10

### PR DESCRIPTION
**Pull request by nbanmp**
This also made concretion logic slightly simpler.

### CODE REVIEW

**Comment by nbanmp** [comment]
Someone with authority over circleci: rerun the test cases once z3 is fixed please.

**Comment by nbanmp** [comment]
@norhh It should be in review. I don't think I have the authority to update the labels.

File: mythril/analysis/solver.py [hunk 1]
```diff
@@ -82,39 +84,28 @@ def get_transaction_sequence(global_state, constraints):
 
             minimize.append(transaction.call_data.calldatasize)
 
-            txs[tx_id] = tx_template.copy()
+            concrete_transactions[tx_id] = tx_template.copy()
 
         else:
             creation_tx_ids.append(tx_id)
 
     model = get_model(tx_constraints, minimize=minimize)
```
[the remark this answers is not in the document]
Comment (nbanmp) [reply mythril/analysis/solver.py]: Currently, get_transaction_sequence is only called from modules in locations where the exception thrown is caught and properly handled.

[the remark this answers is not in the document]
Comment (norhh) [reply mythril/analysis/solver.py]: This exception isn't being caught in integer module for integer overflow function, can you check that out?

[the remark this answers is not in the document]
Comment (nbanmp) [reply mythril/analysis/solver.py]: Good catch! I'm on it.

**Pull request merged**